templates/tfengine/components/cicd/README.md

```diff
@@ -1,21 +1,134 @@
-# Continuous Integration (CI) and Continuous Deployment (CD) component
+# Continuous integration (CI) and continuous deployment (CD)
 
-This directory contains the Terraform configs for the Continuous Integration (CI)
-and Continuous Deployment (CD) component.
+This directory defines the resources that need to be deployed manually to set up
+CICD pipelines of Terraform configs.
 
-* The [manual/](./manual) directory contains the Terraform resources that need to
-  be deployed manually to provision the initial CI/CD resources.
+The CI and CD pipelines use
+[Google Cloud Build](https://cloud.google.com/cloud-build) and
+[Cloud Build triggers](https://cloud.google.com/cloud-build/docs/automating-builds/create-manage-triggers)
+to detect changes in the repo, trigger builds, and run the workloads.
 
-* The [auto/](./auto) directory contains the Terraform resouces for CI/CD purposes
-  that are also managed by CI/CD itself.
+## Setup
 
-  Sensitive resources in the devops project such as
+1. In the Terraform Engine config, add a `cicd` block under the `devops` recipe
+    and specify the following attributes:
 
-  * Terraform state bucket
-  * IAM bindings
-  * Cloud Build Triggers
+    * `github`: Configuration block for GitHub Cloud Build triggers, which
+        supports:
+        * `owner`: GitHub repo owner
+        * `name`: GitHub repo name
+    * `cloud_source_repository`: Configuration block for Google Cloud Source
+        Repository Cloud Build triggers, which supports:
+        * `name`: Cloud Source Repository repo name. The Cloud Source
+            Repository should be hosted under the devops project.
+    * `branch_regex`: Regex of the branches to set the Cloud Build Triggers to
+        monitor.
+    * `enable_continuous_deployment`: Whether or not to enable continuous
```

Comment:
I forgot to update this part with the new `triggers` block. I think since we treat the schema as the SoT. Let's remove the detailed description of each field here, i.e. L16-39, and just make a reference to the CICD recipe's README or schema file whichever is used in other places.

Reply:
Done. Given this will be generated after the engine has run the user already knows/calls the cicd recipe, so I didn't add an explicit reference.